node-ui/src/hooks/useStarknet.tsx

```diff
@@ -0,0 +1,295 @@
+import { useCallback, useState } from 'react';
+import { randomBytes } from 'crypto';
+import { getOrCreateKeypair } from '../auth/ed25519';
+
+import apiClient from '../api';
+import {
+  EthSignatureMessageMetadata,
+  LoginRequest,
+  LoginResponse,
+  NodeChallenge,
+  Payload,
+  SignatureMessage,
+  SignatureMessageMetadata,
+  SignData,
+  WalletMetadata,
+  WalletSignatureData,
+} from '../api/dataSource/NodeDataSource';
+import { ResponseData } from '../api/response';
+import { useNavigate } from 'react-router-dom';
+import { setStorageNodeAuthorized } from '../auth/storage';
+import { useServerDown } from '../context/ServerDownContext';
+import { getWalletType } from '../utils/starknetWalletType';
+import { getStarknet, StarknetWindowObject } from 'get-starknet-core';
+import { constants, Signature } from 'starknet';
+import translations from '../constants/en.global.json';
+
+interface LoginProps {
+  setErrorMessage: (msg: string) => void;
+  isLogin: boolean;
+}
+
+interface RequestNodeDataProps {
+  setErrorMessage: (msg: string) => void;
+}
+
+interface useStarknetReturn {
+  login: ({ isLogin, setErrorMessage }: LoginProps) => void;
+  walletLogin: (walletType: string, setErrorMessage: (msg: string) => void) => void;
+  ready: boolean;
+  starknetInstance: StarknetWindowObject | null;
+  argentXId: string;
+  walletSignatureData: WalletSignatureData | null;
+  signData: SignData | null;
+  signMessage: (setErrorMessage: (msg: string) => void) => void;
+  logout: (setErrorMessage: (msg: string) => void) => void;
+  requestNodeData: ({ setErrorMessage }: RequestNodeDataProps) => void;
+  changeMetamaskNetwork: (networkId: string, setErrorMessage: (msg: string) => void) => void;
+}
+
+const t = translations.useStarknet;
+
+export function useStarknet(): useStarknetReturn  {
+  const [starknetInstance, setStarknetInstance] =
+    useState<StarknetWindowObject | null>(null);
+  const [walletSignatureData, setWalletSignatureData] =
+    useState<WalletSignatureData | null>(null);
+  const [signData, setSignData] = useState<SignData | null>(null);
+  const [ready, setReady] = useState<boolean>(true);
+  const { showServerDownPopup } = useServerDown();
+  const navigate = useNavigate();
+
+  const argentXId = 'argentX';
+
+  const walletLogin = useCallback(async (walletType: string, setErrorMessage: (msg: string) => void) => {
+    try {
+      setReady(false);
+      const starknetInstance = getStarknet();
+      if (starknetInstance) {
+        if (walletType === argentXId) {
+          if(window.starknet_argentX) {
+            await starknetInstance.enable(window.starknet_argentX);
+            const wallets: StarknetWindowObject[] =
+            await starknetInstance.getAvailableWallets();
+            const argentX: StarknetWindowObject = wallets.find(
+              (wallet: any) => wallet.id === argentXId,
+            ) as StarknetWindowObject;
+            setStarknetInstance(argentX);
+          }else {
```

Comment:
Lint will parse after commit so you need to check again formated files and commit again